File: lang/python/src/fun/vac/algorithms/sorting/SelectionSort.py
def sort(arr: list) -> None:
    if len(arr) == 0:
        return

    iMin = 0

    bound = len(arr) - 1
    for i in range(bound):
        iMin = i
        for j in range(i + 1, bound + 1):
            if arr[j] < arr[iMin]:
                iMin = j
        if iMin != i:
            arr[i], arr[iMin] = arr[iMin], arr[i]

File: lang/ruby/src/fun/vac/algorithms/sorting/SelectionSort.rb
module SelectionSort

    def self.sort(arr)
        if arr.empty?
            return
        end

        iMin = 0

        bound = arr.length - 1
        (0 ... bound).each do |i|
            iMin = i
            (i + 1 .. bound).each do |j|
                if arr[j] < arr[iMin]
                    iMin = j
                end
            end
            if iMin != i
                arr[i], arr[iMin] = arr[iMin], arr[i]
            end
        end
    end
end
